src/herder/HerderTests.cpp

```diff
@@ -727,6 +726,52 @@ TEST_CASE("whitelist", "[herder]")
         REQUIRE(txSet->checkValid(*app));
     }
 
+    SECTION("all whitelisted")
+    {
+
+        DataValue value;
+        value.resize(4);
+        SignatureHint hint = SignatureUtils::getHint(accountWL.getPublicKey().ed25519());
+        for (int n = 0; n < 4; n++)
+        {
+            value[n] = (unsigned char)hint[n];
+        }
+
+        whitelist.manageData(KeyUtils::toStrKey(accountWL.getPublicKey()),
+                             &value);
+
+        closeLedgerOn(*app, 2, 4, 11, 2018, txSet->mTransactions);
```

Comment:
why is there a hard coded date here? if its intentionally set to the past, lets make it clear with a comment. maybe also use the epoch (1/1/1970)

Reply:
It's hard-coded because the helper function being invoked requires it.  The date chosen was probably the current date when Or wrote his first unit test.  Looking at other callers, the chosen date seems arbitrary, with successive calls within the same test using successive dates.

Reply:
More data:

The date is converted into a timestamp, with a hard-coded time of midnight.  This is why successive closings in the same test must be on different dates.

In addition, the genesis ledger for tests is hardcoded to 1/7/2014, which sets a lower bound for valid dates, but does not impose an upper bound.